src/routes/api/webhooks/github.ts

```diff
@@ -0,0 +1,300 @@
+import * as crypto from 'node:crypto'
+import { MessageEmbed } from 'discord.js'
+
+function createReleaseMessage(payload) {
+  let embed = new MessageEmbed()
+  embed.setTitle(`[${payload.repository.full_name}] ${payload.release.name}`)
+  embed.setColor('#ff9900')
+  embed.setDescription(payload.release.body)
+  embed.setURL(payload.release.html_url)
+  embed.setAuthor({
+    name: payload.repository.full_name,
+    url: payload.release.author.html_url,
+    iconURL: payload.release.author.avatar_url,
+  })
+  return {
+    content: 'New Release!',
+    tts: false,
+    embeds: [embed],
+  }
+}
+
+// https://gist.github.com/stigok/57d075c1cf2a609cb758898c0b202428
+function verifyGithubWebhookEvent(payloadbody, signature256: string) {
+  const token = process.env.GITHUB_WEBHOOK_SECRET
+  const sig = Buffer.from(signature256 || '', 'utf8')
+  const hmac = crypto.createHmac('sha256', token)
+  const digest = Buffer.from(
+    'sha256' + '=' + hmac.update(JSON.stringify(payloadbody)).digest('hex'),
+    'utf8'
+  )
+  if (sig.length !== digest.length || !crypto.timingSafeEqual(digest, sig)) {
+    throw new Error('Invalid webhook event')
+  }
+  return true
+}
+
+export async function post({ request }) {
+  const payload = await request.json()
+
+  if (!import.meta.vitest) {
+    const sig256 = request.headers.get('x-hub-signature-256')
+    verifyGithubWebhookEvent(payload, sig256)
+  }
+
+  const message = createReleaseMessage(payload)
+
+  const res = await fetch(process.env.DISCORD_WEBHOOK_URL_RELEASES, {
+    headers: {
+      Authorization: `bot ${process.env.DISCORD_BOT_TOKEN}`,
+      'Content-Type': 'application/json',
+    },
+    method: 'POST',
+    body: JSON.stringify(message),
+  })
+
+  // if response is not okay or if Discord did not return a 204
+  // https://discord.com/developers/docs/topics/opcodes-and-status-codes#http
+  if (!res.ok || res.status !== 204) {
```

Comment:
I think just checking `if (!res.ok)` should be sufficient. Just thinking in case Discord changes their API, we can be pretty confident it did what we wanted if the status is anywhere between 200-299.

Reply:
In the error case, should we log the response body for our own debugging purposes?